src/metabase/models/setting.clj

```diff
@@ -467,28 +470,68 @@
 
 (def ^:private ^:dynamic *disable-cache* false)
 
+(def ^:private ^:dynamic *disable-init* false)
+
+(declare get)
+
+(declare set!)
+
+(defn read-setting
+  "Fetch the value of `setting-definition-or-name`. What this means depends on the Setting's `:getter`; by default, this
+  looks for first for a corresponding env var, then checks the cache, then returns the default value of the Setting,
+  if any.
+
+  Note: This will bypass initialization, i.e. it could return nil for a nonce"
+  [setting-definition-or-name]
+  (binding [*disable-init* true]
+    (get setting-definition-or-name)))
+
+(defn- db-value [setting-definition-or-name]
+  (t2/select-one-fn :value Setting :key (setting-name setting-definition-or-name)))
+
+(defn- call-me-maybe [f]
+  (when f (f)))
+
+(defn- db-is-set-up? []
+  ;; this should never be hit. it is just overly cautious against a NPE here. But no way this cannot resolve
+  (call-me-maybe (requiring-resolve 'metabase.db/db-is-set-up?)))
```

Comment:
Maybe you have reasons for changing this, but I found it clearer as it was written before, mostly because it doesn't use a newly created function. Obviously subjective but my taste is to inline new short functions like this because the overhead of reading `call-me-maybe` and wondering what it does is not worth the abstraction it provides.
```suggestion
(def ^:private db-is-set-up?
  (or (requiring-resolve 'metabase.db/db-is-set-up?)
      ;; this should never be hit. it is just overly cautious against a NPE here. But no way this
      ;; cannot resolve
      (constantly false)))
```

Reply:
Just a matter of preference. I like the one liner, but I'll revert if you think it's obtuse. Maybe metabase.utils just needs a better named version of this helper - or has one already!

Reply:
Oh, actually this wasn't just preference - I need to put this inside another function to avoid a circular dependency. This ns is a transitive dependency of metabase.db.

Reply:
Ah understood. In that case here's another suggestion. Feel free to leave it though
```
(cond-> (requiring-resolve 'metabase.db/db-is-set-up?) apply)
```

Reply:
Somehow `apply` was the function I couldn't find - forgot it had a 1-arity 😆

Reply:
1. The `cond->` needs a separate clause predicate, so I think this would end up like:

```clojure
(let [f (requiring-resolve 'metabase.db/db-is-set-up?)]
  (cond-> f f apply))
```

There is `mu/maybe?` which is almost a shorthand for removing the repetition, but it'll return true if `f` is nil.

2. It turns out that `apply` does *not* have a 1-arity 😢 - so this would be `(cond-> f f (apply nil))`, taking even more mental gymnastics to read IMHO.

Personally I find using a helper function clearer - would giving it a better name and sticking it in utils perhaps assuage you? Something like `mu/call-if-some`. Other name ideas are `apply-maybe` or `some-apply`.

For the time being I'm just going to inline to helper at the two call sites 😄